deptool.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/python
+# Copyright (c) 2018 SUSE LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pprint import pprint
+import os
+import sys
+import re
+import logging
+import cmdln
+
+from fnmatch import fnmatch
+from ConfigParser import SafeConfigParser
+import solv
+import rpm
+
+logger = logging.getLogger()
+
+REASONS = [i[14:] for i in dir(solv.Solver) if i.startswith('SOLVER_REASON_')]
+
+class DepTool(cmdln.Cmdln):
```

Comment:
At some point we really should stop introducing more non-osc plugin tools that use the deprecated `cmdln` in favor of `argparse`.

Reply:
yeah, the cmdln interface with the decorators is quite nice though